keps/sig-network/2004-topology-aware-routing/README.md

```diff
@@ -0,0 +1,553 @@
+  - [Prefer](#prefer)
+    - [Terminology](#terminology-1)
+    - [Scheme Logic](#scheme-logic-2)
+    - [Evaluation Tool](#evaluation-tool)
+    - [Test Results](#test-results)
+  - [Recommended Algorithm: Local](#recommended-algorithm-local)
+    - [Optimizations](#optimizations)
+    - [Distribution of EndpointSlices](#distribution-of-endpointslices)
+    - [Rationale](#rationale)
+    - [Potential Issues](#potential-issues)
+  - [Other Algorithms Evaluated](#other-algorithms-evaluated)
+    - [Local-Shared](#local-shared)
+      - [Rationale](#rationale-1)
+      - [Potential Issues](#potential-issues-1)
+    - [Local-Weighted](#local-weighted)
+      - [Rationale](#rationale-2)
+      - [Potential Issues](#potential-issues-2)
+    - [Local-Opt](#local-opt)
+      - [Rationale](#rationale-3)
+      - [Potential Issues](#potential-issues-3)
+  - [Test Plan](#test-plan)
+  - [Observability](#observability)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubernetes clusters are increasingly deployed in multi-zone environments but
+unfortunately network routing has not caught up with that. This doc proposes an
+automatic topology aware routing mechanism with a new Service annotation. This
+new annotation will provide three options of different routing options.
+Currently our proposal will only focus on topology aware routing at zone level.
+
+## Motivation
+
+Kubernetes clusters are increasingly deployed in multi-zone environments.
+Network traffic is routed randomly to any endpoint matching a Service. Some
+users might want the traffic to stay in the same zone for the following
+reasons:
+  - Cost savings: Keeping traffic within a zone can limit cross-zone networking
+costs.
+  - Performance: Traffic within a zone is often more performant than traffic
+leaving the zone.
```

Comment:
nit 
```suggestion
  - Performance: Traffic within a zone usually has less latency and bandwidth contraints, having a better performance than traffic leaving the zone.
```

Reply:
Thanks for the detailed description! Updated.